### PR TITLE
support OpenFile as action type

### PR DESCRIPTION
**Pull request by NetaMolcho**

### CODE REVIEW

File: backend/src/types/GuidedDev.ts [hunk 1]
```diff
@@ -47,6 +47,10 @@ export interface ISnippetAction extends IAction {
     snippet: ISnippet;
 }
 
+export interface IOpenFileAction extends IAction {
```
Comment (tomer-epstein) [inline backend/src/types/GuidedDev.ts:50]: export interface IFileAction extends IAction {

File: backend/src/types/GuidedDev.ts [hunk 2]
```diff
@@ -65,3 +69,7 @@ export interface ISnippet {
     snippetName: string;
     context: any;
 }
+
+export interface IOpenFile {
```
Comment (tomer-epstein) [inline backend/src/types/GuidedDev.ts:73]: IFile

File: backend/src/types/GuidedDev.ts [hunk 3]
```diff
@@ -47,6 +47,10 @@ export interface ISnippetAction extends IAction {
     snippet: ISnippet;
 }
 
+export interface IOpenFileAction extends IAction {
+    file: IOpenFile;
```
Comment (tomer-epstein) [inline backend/src/types/GuidedDev.ts:51]: file: IFile;

File: backend/src/vscode-events.ts [hunk 4]
```diff
@@ -14,6 +14,9 @@ export class VSCodeEvents implements AppEvents {
             case ActionType.Snippet:
               let snippetAction = (action as ISnippetAction);
               return vscode.commands.executeCommand("loadCodeSnippet", {contributorId: snippetAction.snippet.contributorId, snippetName: snippetAction.snippet.snippetName, context: snippetAction.snippet.context});
+            case ActionType.File:
+              let openFileAction = (action as IOpenFileAction);
```
Comment (tomer-epstein) [inline backend/src/vscode-events.ts:18]: let fileAction = (action as IFileAction);

File: backend/src/types/GuidedDev.ts [hunk 5]
```diff
@@ -65,3 +69,7 @@ export interface ISnippet {
     snippetName: string;
     context: any;
 }
+
+export interface IOpenFile {
+    uri: string;
```
Comment (tomer-epstein) [inline backend/src/types/GuidedDev.ts:74]: uri: vscode.Uri;

File: backend/src/vscode-events.ts [hunk 6]
```diff
@@ -14,6 +14,9 @@ export class VSCodeEvents implements AppEvents {
...
+              return vscode.commands.executeCommand('vscode.open', vscode.Uri.parse(openFileAction.file.uri));
```
Comment (tomer-epstein) [inline backend/src/vscode-events.ts:19]: return vscode.commands.executeCommand('vscode.open', fileAction.file.uri);

File: vscode-contrib3/src/extension.ts [hunk 7]
```diff
@@ -294,5 +334,14 @@ function getImage(imagePath: string) :string {
     return image;
 }
 
+function getFileFromWorkspace(fileName: string) : string {
```
Comment (tomer-epstein) [inline vscode-contrib3/src/extension.ts:337]: getFileUriFromWorkspace

**Review by tomer-epstein — CHANGES_REQUESTED** [review]

File: backend/tests/vscode-events.spec.ts [hunk 8]
```diff
@@ -106,8 +106,11 @@ describe('vscode-events unit test', () => {
             }
             return events.performAction(item, 1);
         });
-        it("File as ActionType", () => { 
-            uriMock.expects("parse").withExactArgs("https://chakra-ui.com/alert").resolves();
+        it.skip("File as ActionType", () => { 
+            uriMock.expects("parse").withExactArgs("README").resolves();
+
```
Comment (tomer-epstein) [inline backend/tests/vscode-events.spec.ts]: This doesn't test anything in our code --> uriMock.expects("parse") 
why is it in skip?

File: backend/tests/vscode-events.spec.ts [hunk 9]
```diff
@@ -150,4 +153,28 @@ describe('vscode-events unit test', () => {
             return events.performAction(item, 1);
         });
     });
+    describe("performAction - on failure", () => {
+        it("action or actionType does not exist", () => {
+            commandsMock.expects("executeCommand").never();
+            uriMock.expects("parse").never();
```
Comment (tomer-epstein) [inline backend/tests/vscode-events.spec.ts:156]: ??

File: vscode-contrib3/src/extension.ts [hunk 10]
```diff
@@ -334,13 +314,14 @@ function getImage(imagePath: string) :string {
     return image;
 }
 
-function getFileFromWorkspace(fileName: string) : string {
+function getFileUriFromWorkspace(fileName: string) :vscode.Uri {
     let outputFolder = _.get(vscode, "workspace.workspaceFolders[0].uri.path");
```
Comment (tomer-epstein) [inline vscode-contrib3/src/extension.ts:318]: let outputFolderUri = _.get(vscode, "workspace.workspaceFolders[0].uri");

File: vscode-contrib3/src/extension.ts [hunk 11]
```diff
@@ -334,13 +314,14 @@ function getImage(imagePath: string) :string {
...
     if (!outputFolder || !outputFolder.length) {
         vscode.window.showErrorMessage("Cannot find folder");
-        return "";
+        return vscode.Uri.parse("");;
     }
-    return outputFolder + '/' + fileName;
+    outputFolder = outputFolder + '/' + fileName;
+    return vscode.Uri.parse(outputFolder);
```
Comment (tomer-epstein) [inline vscode-contrib3/src/extension.ts:324]: return outputFolderUri;

File: backend/tests/vscode-events.spec.ts (hunk 8 above)
Comment (NetaMolcho) [reply backend/tests/vscode-events.spec.ts]: this is Outdated, you need to refresh the pr

File: backend/tests/vscode-events.spec.ts [hunk 12]
```diff
@@ -43,6 +44,7 @@ describe('vscode-events unit test', () => {
         events = new VSCodeEvents();
         windowMock = sandbox.mock(vscode.window);
         commandsMock = sandbox.mock(vscode.commands);
+        uriMock = sandbox.mock(vscode.Uri);
```
Comment (tomer-epstein) [inline backend/tests/vscode-events.spec.ts:47]: no need

File: backend/tests/vscode-events.spec.ts [hunk 13]
```diff
@@ -51,6 +53,7 @@ describe('vscode-events unit test', () => {
         windowMock.verify();
         eventsMock.verify();
         commandsMock.verify();
+        uriMock.verify();
```
Comment (tomer-epstein) [inline backend/tests/vscode-events.spec.ts:56]: no need

**Review by tomer-epstein — APPROVED** [review]

**Pull request merged**